src/App.tsx

```diff
@@ -1,60 +1,151 @@
 import './App.scss';
+import { useEffect, useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
 export const App = () => {
+  const [titleInput, setTitleInput] = useState('');
+  const [userSelect, setUserSelect] = useState('');
+  const [checkInput, setCheckInput] = useState(false);
+  const [showSpanTitle, setShowSpanTile] = useState(false);
+  const [showSpanUser, setShowSpanUser] = useState(false);
+  const [visibleTodos, setVisibleTodos] = useState([...todosFromServer]);
+
+  useEffect(() => {
+    if (titleInput.length !== 0 && +userSelect !== 0) {
+      setCheckInput(true);
+    }
+
+    if (titleInput.length !== 0) {
+      setShowSpanTile(false);
+    }
+
+    if (+userSelect !== 0) {
+      setShowSpanUser(false);
+    } else {
+      setCheckInput(false);
+    }
+  });
+
+  const addTodo = () => {
+    if (checkInput) {
+      const id = Math.max(...visibleTodos.map(todo => todo.id));
+
+      const newTodo = {
+        id: id + 1,
+        title: titleInput,
+        userId: +userSelect || 0,
+        completed: false,
+      };
+
+      // return visibleTodos;
+      setVisibleTodos([
+        ...visibleTodos, newTodo,
+      ]);
+
+      setTitleInput('');
+      setUserSelect('0');
+    }
+
+    if (titleInput.length === 0) {
+      setShowSpanTile(true);
+    }
+
+    if (+userSelect === 0) {
+      setShowSpanUser(true);
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
       <form action="/api/users" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            value={titleInput}
+            onChange={(event) => {
+              setTitleInput(event.target.value);
+            }}
+          />
+          {
+            showSpanTitle
+              ? <span className="error">Please enter a title</span>
+              : <></>
+          }
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          <select
+            data-cy="userSelect"
+            value={userSelect || '0'}
```

Comment:
It'll be 0 by default if you correct the comment above
```suggestion
            value={userSelect}
```

Reply:
not fixed yet